File: MyArray.java
import java.util.*;

class MyArray {
   static final int MAX = 100;
   int[] array;
   boolean duplicate;

   MyArray(int size) {
      this(size, true);
   }

   MyArray(int size, boolean duplicate) {
      this.array = new int[size];
      this.duplicate = duplicate;
   }

   void generate() {
      Random random = new Random();
      if (duplicate) {
         for (int i = 0; i < array.length; i++) {
            int number = random.nextInt(MAX);
            array[i] = number;
         }
      } else {
         // no duplicate allowed!
         Set<Integer> set = new HashSet<>();
         // generate numbers into a set, to avoid duplicates
         while (set.size() != array.length) {
            int number = random.nextInt(MAX);
            set.add(number);
         }
         // transfer the set contents to the data array
         Iterator<Integer> it = set.iterator();
         int i = 0;
         while (it.hasNext()) {
            array[i] = it.next();
            i++;
         }
      }
   }

   public String toString() {
      return toString(array, 0, array.length);
   }

   String toString(int[] array, int low, int high) {
      StringBuilder builder = new StringBuilder();
      for (int i = low; i < high; i++) {
         builder.append(array[i]).append(" ");
      }
      return builder.toString();
   }
}

File: MyArray.cpp
#include <cstdlib>
#include <unordered_set>
#include "MyArray.h"

#define MAX 100;

MyArray::MyArray(int size, bool duplicate) {
   this->array = new int[size];
   this->size = size;
   this->duplicate = duplicate;
}

void MyArray::generate() {
   srand(time(NULL));
   if (duplicate) {
      for (int i = 0; i < size; i++) {
         int number = rand() % MAX;
         array[i] = number;
      }
   } else {
      // no duplicate allowed!
      // note the use of unordered_set and not set, because set sorts its contents
      unordered_set<int> tmp;
      // generate numbers into a set, to avoid duplicates
      while (tmp.size() != size) {
         int number = rand() % MAX;
         tmp.insert(number);
      }
      // transfer the set contents to the data array
      int i = 0;
      for (unordered_set<int>::iterator it = tmp.begin(); it != tmp.end(); it++) {
         array[i] = *it;
         i++;
      }
   }
}

ostream& operator<<(ostream& stream, MyArray& array) {
   for (int i = 0; i < array.size; i++)
      stream << array.array[i] << " ";
   return stream;
}
